crypto/elytron/src/main/java/org/keycloak/crypto/elytron/ElytronCertificateUtils.java

```diff
@@ -249,16 +252,26 @@ public List<String> getCRLDistributionPoints(X509Certificate cert) throws IOExce
                 case ASN1.UTF8_STRING_TYPE:
                    distPointUrls.add(der.decodeUtf8String());
                    break;
-                case 0xa0:
-                   der.decodeImplicit(0xa0);
-                   byte[] edata = der.decodeOctetString();
-                   while(!Character.isLetterOrDigit(edata[0])) {
-                    edata = Arrays.copyOfRange(edata, 1, edata.length);
-                }
-                   distPointUrls.add(new String(edata));
+                case 0xa0: // Decode CRLDistributionPoint FullName list
+                   der.startExplicit(0xa0);
+                   break;
+                case 0x86: // Decode CRLDistributionPoint FullName
+                   der.decodeImplicit(0x86);
+                   distPointUrls.add(der.decodeOctetStringAsString());
+                   log.debug("Adding Dist point name: " + distPointUrls.get(distPointUrls.size()-1));
                    break;
                 default:
                    der.skipElement();
+            }
+            // Check to see if there is another sequence to process
+            try {
+                if(!der.hasNextElement() && der.peekType() == ASN1.SEQUENCE_TYPE) {
```

Comment:
Would mind explaining why we are expecting exceptions and ignoring them here?

Not sure if I'm following the call to decode a sequence if there are no elements to process?

Reply:
I added a log to at least print out the exception. As I was testing, I would run into an issue where this check for additional elements would throw an exception when there weren't any more elements. I eventually figured out what I was doing wrong in those cases. But, I left the try catch block to handle it if other cert encodings cause the same issue.